### DISCUSSION TITLE
does latest openssl plugin still support OpenSSL 1.1.1 and how to specify the included version

### DISCUSSION THREAD

**Discussion by Incises** [strongswan/strongswan#2116] [Q&A]
I have configured with `--enable-openssl` to use the [openssl plugin](https://github.com/strongswan/strongswan/blob/master/src/libstrongswan/plugins/openssl/openssl_plugin.c), and the latest release works fine on my computer. When I look up the dynamic library dependencies of `libstrongswan` and `libcharon`, there isn't any link to `libcrypto`.

However, I need to build strongSwan with OpenSSL 1.1.1 to support some old client machines. 
With both version (openssl@1.1 and openssl@3) installed in my building environment, how can I use the former one? 
Should I do it by specifying `CFLAGS` and `LDFLAGS` environment variables?

**Comment by tobiasbrunner** [comment]
> When I look up the dynamic library dependencies of `libstrongswan` and `libcharon`, there isn't any link to `libcrypto`.

If you don't build with `--enable-monolithic` only the `libstrongswan-openssl.so` plugin/shared library will have a link to `libcrypto`.



> Should I do it by specifying `CFLAGS` and `LDFLAGS` environment variables?

Yep.
